cloud/store/recover.go

```diff
@@ -0,0 +1,208 @@
+				continue
+			}
+
+			var peer Peer
+			if address, ok := serverMap["Address"].(string); ok {
+				peer.Address = address
+			}
+			if id, ok := serverMap["ID"].(string); ok {
+				peer.ID = id
+			}
+			if suffrage, ok := serverMap["Suffrage"].(int64); ok {
+				peer.Suffrage = suffrage
+				peer.NonVoter = suffrage != int64(raft.Voter)
+			}
+
+			existedP := peersMap[peer.Address]
+			if reflect.DeepEqual(existedP, peer) {
+				continue
+			}
+			peersMap[peer.Address] = peer
+			peers = append(peers, peer)
+		}
+	}
+
+	if len(peers) == 0 {
+		return nil
+	}
+
+	configData, err := json.MarshalIndent(peers, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(peerFilePath, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0o644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	if _, err := file.Write(configData); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// configViaPeers checks if the peers.json file exists with the provided path
+// if the files exists it creates Raft.Configuration from it
+func (st *Store) configViaPeers(peerFilePath string) (raft.Configuration, error) {
+	var configuration raft.Configuration
+	if _, err := os.Stat(peerFilePath); err != nil {
+		st.log.Error("either file doesn't exist, or there an error", err)
+		// if files doesn't exist we bail out early
+		return configuration, nil
+	}
+
+	st.log.Info(fmt.Sprintf("found %s file, collect peers from previous config...", peerFilePath))
+
+	configuration, err := raft.ReadConfigJSON(peerFilePath)
+	if err != nil {
+		return configuration, fmt.Errorf("raft recovery failed to parse peers.json: %w", err)
+	}
+	if err := os.RemoveAll(peerFilePath); err != nil {
+		return configuration, fmt.Errorf("recovery failed to delete %s, please delete manually ,err=%w", peerFilePath, err)
+	}
+
+	st.log.Info(fmt.Sprintf("deleted %s file and will proceed with the recovery", peerFilePath))
+
+	st.log.Info("previous raft", "peers", fmt.Sprintf("%v", configuration.Servers))
+	return configuration, nil
+}
+
+// TODO-WEAVIATE-680: at the moment the validation of the provided address happens on the ip level but not ports,
+// improvement would be validate also that the provided ip and ports exists in the memberlist.
+func (st *Store) recoverable(peers []raft.Server) ([]raft.Server, bool) {
+	if len(peers) == 0 {
+		st.log.Warn("provided server list is empty, recovery is not possible")
+		return nil, false
+	}
+
+	var aliveServes []raft.Server
+	serversMap := make(map[string]bool, len(peers))
+	// sleep to wait for members to join memberlist
+	for {
+		select {
+		case <-time.After(st.recoveryTimeout):
+			return aliveServes, len(aliveServes) > 0
+		case <-time.After(1 * time.Second):
+			st.log.Info("existing members in memberlist", "IPs", fmt.Sprintf("%v", st.cluster.AllHostnames()))
+			for _, s := range peers {
+				ip := strings.Split(string(s.Address), ":")[0]
+				// remove not alive IPS
+				if !st.cluster.Alive(ip) {
```

Comment:
I believe, we need to find the IP address of a server given its ID rather its old IP. 
The old IP could be attributed to another node in the cluster or it doesn't exit any more.

Also we probably need to wait a little bit longer until more nodes join, otherwise, we risk only the local node being present in the returned list